entities.py

```diff
@@ -64,6 +65,54 @@ def distance_to(self, dest):
         return (self.pos - dest).norm
 
 
+class Item(Entity):
+
+    def __init__(self, name, pos, char, color, blocks, effect=None):
+        super().__init__(name, pos, 'item', char, color, blocks, RenderPriority.ITEM)
+        self.behavior = NullBehavior()
+        self.effect = effect
+
+    def remove(self):
+        self.char = ' '
+        self.blocks = False
+
+    def use(self, player, target):
+        if self.effect:
+            self.effect(target)
+        else:
+            message("Nothing happened...")
+
+    def take_turn(self, target, game_map):
+        self.behavior.take_turn(None, None, None)
+
+
+class Backpack(metaclass=Singleton):
+
+    contents = OrderedDict()
+
+    def _search(cls, name):
+        for item in cls.contents:
+            if item.name == name:
+                return item
+
+    def add(cls, item, qty):
+        c_item = cls._search(item.name)
+        if c_item:
+            cls.contents[c_item] += qty
+        else:
+            if len(cls.contents) < 24:
+                cls.contents[item] = qty
+
+    def use(cls, i, player, target):
```

Comment:
Why pass it `i`? It's kinda confusing. We can do like other roguelikes and assign each item in the backpack a key, which is just a keyboard key (letter or number). With that in mind, we could make a ``UseItemCommand` that takes the key the player pressed and calls this method, which would check if there's an item with that corresponding key and use it.

Reply:
It's the same IMO, whether you pass an index or a key the user will have to look at the UI to know which item he will use, ``i`` being the ith element in the backpack (thus the ``OrderedDict``), also remember that backpack keys are item objects, I didn't want to have a dict within a dict